### PR TITLE
[pioneeravr] Add models up until year 2020

### LINKED ISSUE #7315: [pioneeravr] add 2016+ models
The pioneeravr binding is missing support for AVR models released since 2016.
As the protocol has not changed it is merely a matter of adding products names to a list.

Pinging @gersilex who might be able to author a PR as the original author isn't available.

[PioneerAvrBindingConstants.java.txt](https://github.com/openhab/openhab-addons/files/4440859/PioneerAvrBindingConstants.java.txt)

### PR DESCRIPTION
**Pull request by gersilex**
This improvement of the PioneerAVR binding adds support for the models @mstormi provided under the assumption that
the features remained the same since 2015.
The models' functionality remains split and is
easy to enhance with model-year-specific properties later on.

closes #7315

Also-by: Markus Storm <markus.storm@gmx.net>
Signed-off-by: Leroy Foerster <gersilex@gmail.com>

### CODE REVIEW

**Comment by J-N-K** [comment]
Is this a DRAFT by intention? Please make it a PR so we can review and merge.

**Comment by gersilex** [comment]
Yes, it is intentional as my battery was dying and I was not able to complete my work yesterday :D 

I'll make it a PR and mention you as soon as I am ready and the PR meets the quality requirements.

**Comment by gersilex** [comment]
https://s3.eu-central-1.amazonaws.com/leroy.foerster.it.dienstleistung/org.openhab.binding.pioneeravr-2.5.4-SNAPSHOT.jar

**Comment by gersilex** [comment]
@J-N-K ready to review

**Comment by J-N-K** [comment]
Why is so much channel definition copied? The input-source definition looks the same for some years, so it would be possible to re-use it (and that would also allow re-using the channel-group-type.

And why is the channel-group labelled with a year ( I see that this is also the case for older models, but isn't that unnecessary? The receiver model should already define the year, so it is a duplicate on the channel-group label, right?

**Comment by gersilex** [comment]
@J-N-K I agree to both your points. I found it weird and duplicated, too. I thought about combining the models without years and the models *with* years as they seem to have the same channels every time in a different GitHub Issue. 

On the other hand, I can not be sure that all models have the same features and extending it with more zones or different input types will be much easier if we have the year/generation split.

tl;dr I agree and I don't have knowledge about all the models and their feature sets or the time to dig deep into their documentation. Combining the non-year and the year models is acceptable though and I can do it, if you guys consider it better. It will require actions on the users though, as the thing type will change.

**Comment by cpmeister** [comment]
Until the thing xml files allow for inheritance, duplication is unavoidable.
